### PR TITLE
[AST] Include clang/Basic/DiagnosticComment.h instead of clang/AST/CommentDiagnostic.h

### PR DESCRIPTION
**Pull request by kazutakahirata**
Since:

  commit d076608d58d1ec55016eb747a995511e3a3f72aa
  Author: Richard Trieu <rtrieu@google.com>
  Date:   Sat Dec 8 05:05:03 2018 +0000

clang/AST/CommentDiagnostic.h has been forwarding to
clang/Basic/DiagnosticComment.h.  This patch includes
clang/Basic/DiagnosticComment.h instead of
clang/AST/CommentDiagnostic.h.

### CODE REVIEW

File: clang/include/module.modulemap [hunk 1]
```diff
@@ -111,7 +111,6 @@ module Clang_Diagnostics {
   module All { header "clang/Basic/AllDiagnostics.h" export * }
   module Analysis { textual header "clang/Analysis/Analyses/UnsafeBufferUsageGadgets.def" }
   module AST { header "clang/AST/ASTDiagnostic.h" export * }
-  module Comment { header "clang/AST/CommentDiagnostic.h" export * }
```
Comment (Sirraide) [inline clang/include/module.modulemap:114]: Do we want to replace this with a reference to `DiagnosticComment.h` instead of removing it entirely?

**Review by Sirraide — APPROVED** [review]
Yeah, I think we can remove this header at this point.

I do have a question about the modulemap though.

File: clang/include/module.modulemap (hunk 1 above)
Comment (AaronBallman) [reply clang/include/module.modulemap:114]: CC @ChuanqiXu9 @Bigcheese

**Review by AaronBallman — APPROVED** [review]
LGTM modulo the module question.

File: clang/include/module.modulemap (hunk 1 above)
Comment (kazutakahirata) [reply clang/include/module.modulemap:114]: I am guessing no, but I admit I am no modulemap expert though.

`module.modulemap` says:

```
module Clang_Diagnostics {
  requires cplusplus

  module All { header "clang/Basic/AllDiagnostics.h" export * }
```

Now, `AllDiagnostics.h` includes `clang/Basic/DiagnosticComment.h`, so `module All` above seems to imply `DiagnosticComment.h`.

**Comment by MaskRay** [comment]
@Weverything :)

**Comment by Weverything** [comment]
I support getting rid of the Diagnostic.h files outside of the Basic directory, at least when they are just forwarding headers.  From a layering point of view, Clang_Diagnostics should be the same level as Basic so it shouldn't be having things from AST.  Removing the forwarding headers will help clean up the layering.

**Pull request merged**